src/software/ai/hl/stp/tactic/dribble/dribble_fsm.cpp

```diff
@@ -106,6 +136,33 @@ void DribbleFSM::dribble(const Update &event)
             event.control_params.dribble_destination,
             event.control_params.final_dribble_orientation);
 
+    Point dribble_destination = getDribbleBallDestination(
+        event.common.world.ball().position(), event.control_params.dribble_destination);
+
+    // when we have the ball in the dribbler, only then we pivot around the ball
+    // if (distance(event.common.robot.position(), event.common.world.ball().position()) <
+    // 0.088)
+    if (event.common.robot.isNearDribbler(event.common.world.ball().position()))
+    {
+        Angle ang_1 = event.common.robot.orientation() + Angle::fromDegrees(40);
```

Comment:
I think the logic in this if statement can be simplified to something like (this may have some bugs in it): 
```cpp
Angle remaining_rotation = (target_orientation - robot_orientation).clamp();
double amount_to_rotate_deg = std::clamp(remaining_rotation.toDegrees(), -40, 40);
Angle target_angle = robot_orientation + Angle::toDegrees(amount_to_rotate_deg);
```
`PrimitiveExecutor::getTargetAngularVelocity` also has some similar logic that you can use for inspiration.

Reply:
Can probably share the same logic with `getPossession` aswell